tests/test_type/test_array2.py

```diff
@@ -393,3 +393,16 @@ class Foo(m.Circuit):
         for i, driving in enumerate(io.I.driving()):
             assert len(driving) == 1
             assert driving[0] is io.O[i]
+
+
+def test_array2_value_return_type():
```

Comment:
can we move this test to somewhere else and make it more specific to the issue at hand? maybe `tests/test_wire/test_introspection.py`.

And test can be as simple as

```python
@pytest.mark.parameterize("T", (m.Bits,))
def test_value_type(T):
    x = m.Out(T(name="x"))
    y = m.In(T(name="y"))
    y @= x
    assert isinstance(y.value(), T)
```

Reply:
Moved, but the test can't be quite that simple.  The issue above only arises when array isn't wired as whole, so it needs to test an array that's had elaborated children.